redisgears_core/src/lib.rs

```diff
@@ -981,7 +982,128 @@ fn js_init(ctx: &Context, _args: &[RedisString]) -> Status {
     Status::Ok
 }
 
-const fn js_info(_ctx: &InfoContext, _for_crash_report: bool) {}
+fn build_uninitialised_backends_info(ctx: &InfoContext) -> RedisResult<()> {
+    if get_uninitialised_backends_mut().is_empty() {
+        return Ok(());
+    }
+
+    let mut section_builder = ctx.builder().add_section("UninitialisedBackends");
+
+    section_builder = get_uninitialised_backends_mut()
+        .keys()
+        .try_fold(section_builder, |section_builder, name| {
+            section_builder.field("backend_name", name.as_str())
+        })?;
+
+    let _ = section_builder.build_section()?.build_info()?;
+
+    Ok(())
+}
+
+fn build_backend_module_info(ctx: &InfoContext, info: ModuleInfo) -> RedisResult<()> {
+    let mut builder = ctx.builder();
+    for section in &info.sections {
+        let mut section_builder = builder.add_section(section.0);
+        match section.1 {
+            InfoSectionData::KeyValuePairs(map) => {
+                for (key, value) in map {
+                    section_builder = section_builder.field(key, value.as_str())?;
+                }
+            }
+            InfoSectionData::Dictionaries(dictionaries) => {
+                for dictionary in dictionaries {
+                    let mut dictionary_builder = section_builder.add_dictionary(dictionary.0);
+                    for (key, value) in dictionary.1 {
+                        dictionary_builder = dictionary_builder.field(key, value.as_str())?;
+                    }
+                    section_builder = dictionary_builder.build_dictionary()?;
+                }
+            }
+        }
+        builder = section_builder.build_section()?;
+    }
+
+    let _ = builder.build_info()?;
+
+    Ok(())
+}
+
+fn build_initialised_backends_info(ctx: &InfoContext) -> RedisResult<()> {
+    get_backends_mut()
+        .values_mut()
+        .filter_map(|b| b.get_info())
+        .try_for_each(|info| build_backend_module_info(ctx, info))
+}
+
+fn build_per_library_info(ctx: &InfoContext) -> RedisResult<()> {
+    let libraries = get_globals().libraries.lock()?;
+    if libraries.is_empty() {
+        return Ok(());
+    }
+
+    let builder = ctx.builder();
+    let section_builder = builder.add_section("PerLibraryInformation");
+    let mut dictionaries = HashMap::new();
+    for library in libraries.iter() {
+        let mut library_info = HashMap::new();
+
+        library_info.insert(
+            "function_count".to_owned(),
+            library.1.gears_lib_ctx.functions.len().to_string(),
+        );
```

Comment:
I suggest maybe to split between function and async functions? Do we have this information?